### PR TITLE
feat: add `.url()`to access the current url

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: src/webview/webview2/mod.rs [hunk 1]
```diff
@@ -767,6 +768,17 @@ window.addEventListener('mousemove', (e) => window.chrome.webview.postMessage('_
     let _ = self.eval("window.print()");
   }
 
+  pub fn url(&self) -> Url {
+    let mut pwstr = PWSTR::null();
+
+    let webview = unsafe { self.controller.CoreWebView2().unwrap() };
+    unsafe { webview.Source(&mut pwstr).unwrap() };
```
Comment (amrbashir) [inline src/webview/webview2/mod.rs]:
```suggestion
    unsafe { self.webview.Source(&mut pwstr).unwrap() };
```

File: src/webview/android/mod.rs [hunk 2]
```diff
@@ -228,6 +229,10 @@ impl InnerWebView {
 
   pub fn print(&self) {}
 
+  pub fn url(&self) -> Url {
+    todo!();
```
Comment (amrbashir) [inline src/webview/android/mod.rs]: this will panic, could it maybe replaced with a default value? or change the function signature to use `Result`

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/webview/android/mod.rs (hunk 2 above)
Comment (JonasKruckenberg) [reply src/webview/android/mod.rs]: My assumption was that we actually want this function on android too, I just don't know how to implement it that's why I marked it as `todo`.

Or is this not possible on android?

Comment (amrbashir) [reply src/webview/android/mod.rs]: No it is possible, let me add it

**Review by amrbashir — APPROVED** [review]

**Review by amrbashir — CHANGES_REQUESTED** [review]
I am once again asking for change file.

**Review by amrbashir — APPROVED** [review]

**Pull request merged**